tvplayer/src/main/java/com/kaltura/tvplayer/KalturaPlayer.java

```diff
@@ -85,67 +87,112 @@
     private PlayerTokenResolver tokenResolver = new PlayerTokenResolver();
     private PlayerInitOptions initOptions;
 
-    public static KalturaPlayer createOVPPlayer(Context context, PlayerInitOptions initOptions) {
-        kalturaPlayerType = KalturaPlayerType.ovp;
-        KalturaPlayer kalturaPlayer = new KalturaPlayer(context, initOptions);
-        kalturaPlayer.serverUrl = KalturaPlayer.safeServerUrl(initOptions.serverUrl, KalturaPlayer.DEFAULT_OVP_SERVER_URL);
-        return kalturaPlayer;
+    public static void initializeOVP(Context context, int partnerId, @Nullable String serverUrl) {
+
+        PlayerConfigManager.retrieve(context, Type.ovp, partnerId, serverUrl, (config, error, freshness) -> {
+           if (error != null) {
+               log.e("initialize KalturaPlayerType failed");
```

Comment:
should we have a way to notify the app that init failed? passing completion handler

Reply:
Might be a good idea.

Reply:
done